src/logic/arrangers/JsonArranger.ts

```diff
@@ -38,40 +44,41 @@ function arrangeJson(data: any, offset = 0): JsonArrangerResult {
 
 function arrangePrimitive(
   data: boolean | number | string | undefined | null,
-  offset = 0,
-): LineContent {
+): InlineContent {
   return {
-    type: 'line',
-    line: offset,
-    children: [
-      {
-        text: typeof data === 'string' ? `"${data}"` : '' + data,
-      },
-    ],
+    text: typeof data === 'string' ? `"${data}"` : '' + data,
   }
 }
 
-function arrangeArray(data: unknown[], offset = 0): JsonArrangerResult {
+function arrangeArray(
+  data: unknown[],
+  offset = 0,
+  indent = 0,
+): JsonArrangerResult {
   const lineBegin = offset
   const children = [
     {
       type: 'line',
       line: offset,
-      children: [{ text: '[' }],
+      children: [createIndent(indent), { text: '[' }].filter(Boolean),
     },
   ] as (LineContent | NestedContent)[]
   offset += 1
 
   for (let index = 0; index < data.length; ++index) {
-    const { content, nextOffset } = arrangeJsonRaw(data[index], offset)
-    children.push(insertComma(content, index === data.length - 1))
+    const { content, nextOffset } = arrangeJsonRaw(
+      data[index],
+      offset,
+      indent + 1,
+    )
+    children.push(insertComma(content, index === data.length - 1, indent + 1))
     offset = nextOffset
   }
 
   children.push({
     type: 'line',
     line: offset,
-    children: [{ text: ']' }],
+    children: [createIndent(indent)!, { text: ']' }].filter(Boolean),
```

Comment:
An alternative looks like this

```ts
[createIndent(indent), { text: ']' }].filter(Boolean) as InlineContent[]
```

and I hope you would know why I didn't do like that.